dbms/src/Interpreters/executeQuery.cpp

```diff
@@ -232,16 +232,8 @@ std::tuple<ASTPtr, BlockIO> executeQueryImpl(
 
             context.setProcessListElement(&process_list_entry->get());
         }
```

Comment:
maybe
```
if (!internal && nullptr == typeid_cast<const ASTShowProcesslistQuery *>(&*ast))
{
    process_list_entry = xxx
}
else if (context.getDAGContext())
{
    process_list_entry = context.getDAGContext()->getProcessListEntry();
}
```
and set the `internal` must be true in https://github.com/pingcap/tiflash/blob/5609e68c0fa213e9740ff52ec60eac97c884e45a/dbms/src/Flash/executeQuery.cpp#L29-L35
is better;

Reply:
or
```
if (context.getProcessListElement())
    process_list_entry = context.getProcessListElement()
else if (!internal && nullptr == typeid_cast<const ASTShowProcesslistQuery *>(&*ast))
{
    xxxxx
}
```

Reply:
I think it is a little hack to set `internal` to be true, and `context.getDAGContext() != nullptr` does not means process_list_entry is not nullptr, only mpp task will construct process_list_entry earlier, for cop/batch cop, it still need to be constructed here.

Reply:
Then maybe we just need to check `if (context.getProcessListElement())` here?

Reply:
But if we just check `if (context.getProcessListElement())`, the reader may be confused because they don't know in which case context.getProcessListElement() will be nullptr?

Reply:
I prefer the orignal implementation because it shows clearly that if current query is a mpp task, it does not need to create process_list_entry here. I can add some comments to explain that mpp task already create process_list_entry so we don't need to create it here.

Reply:
ok, add `assert(context.getProcessListElement());` and some comments is ok to me